app/src/components/v-workspace.vue

```diff
@@ -0,0 +1,168 @@
+	>
+		<div
+			class="workspace"
+			:style="{
+				transform: `scale(${zoomScale})`,
+				width: workspaceSize.width + 'px',
+				height: workspaceSize.height + 'px',
+			}"
+		>
+			<v-workspace-panel
+				v-if="!$slots.panel"
+				v-for="panel in panels"
+				:key="panel.id"
+				v-bind="panel"
+				:edit-mode="editMode"
+				:resizable="resizable"
+				@edit="$emit('edit', panel)"
+				@update="$emit('update', { edits: $event, id: panel.id })"
+				@move="$emit('move', panel.id)"
+				@delete="$emit('delete', panel.id)"
+				@duplicate="$emit('duplicate', panel)"
+			>
+				<slot :panel="panel"></slot>
+			</v-workspace-panel>
+			<template v-else>
+				<template v-for="panel in panels" :key="panel.id">
+					<slot name="panel" :panel="panel"></slot>
+				</template>
+			</template>
+		</div>
+	</div>
+</template>
+
+<script setup lang="ts">
+import { computed, inject, ref } from 'vue';
+import { useElementSize } from '@/composables/use-element-size';
+import { AppPanel } from './v-workspace-panel.vue';
+
+const props = withDefaults(defineProps<{
+	panels: AppPanel[]
+	editMode?: boolean
+	zoomToFit?: boolean
+	resizable?: boolean
+}>(), {
+	editMode: false,
+	zoomToFit: false,
+	resizable: true
+})
+
+defineEmits(['update', 'move', 'delete', 'duplicate', 'edit'])
+
+const mainElement = inject('main-element', ref<Element>());
+const mainElementSize = useElementSize(mainElement);
+
+const paddingSize = computed(() => Number(getVar('--content-padding')?.slice(0, -2) || 0));
+
+const workspaceSize = computed(() => {
+	const furthestPanelX = props.panels.reduce(
+		(aggr, panel) => {
+			if (panel.x! > aggr.x!) {
+				aggr.x = panel.x!;
+				aggr.width = panel.width!;
+			}
+
+			return aggr;
+		},
+		{ x: 0, width: 0 }
+	);
+
+	const furthestPanelY = props.panels.reduce(
+		(aggr, panel) => {
+			if (panel.y! > aggr.y!) {
+				aggr.y = panel.y!;
+				aggr.height = panel.height!;
+			}
+
+			return aggr;
+		},
+		{ y: 0, height: 0 }
+	);
+
+	if (props.editMode === true) {
+		return {
+			width: (furthestPanelX.x! + furthestPanelX.width! + 25) * 20,
+			height: (furthestPanelY.y! + furthestPanelY.height! + 25) * 20,
+		};
+	}
+
+	return {
+		width: (furthestPanelX.x! + furthestPanelX.width! - 1) * 20,
+		height: (furthestPanelY.y! + furthestPanelY.height! - 1) * 20,
+	};
+});
+
+const zoomScale = computed(() => {
+	if (props.zoomToFit === false) return 1;
+
+	const { width, height } = mainElementSize;
+
+	const scaleWidth: number = (width.value - paddingSize.value * 2) / workspaceSize.value.width;
+	const scaleHeight: number = (height.value - 114 - paddingSize.value * 2) / workspaceSize.value.height;
+
+	return Math.min(scaleWidth, scaleHeight);
+});
+
+const workspaceBoxSize = computed(() => {
+	return {
+		width: workspaceSize.value.width * zoomScale.value + paddingSize.value * 2,
+		height: workspaceSize.value.height * zoomScale.value + paddingSize.value * 2,
+	};
+});
+
+function getVar(cssVar: string) {
```

Comment:
Reuse existing util function

Reply:
Reused the existing util function, but since the "scope" was different (currently it uses main element, and it returns 32px. Whereas if we use the existing util function, it returns 12px), I modified the cssVar util function to accept an element as a second parameter and fallback to document.body otherwise. Hope that's in the right direction.